sdks/java/io/mqtt/src/main/java/org/apache/beam/sdk/io/mqtt/MqttIO.java

```diff
@@ -335,8 +339,26 @@ public void finalizeCheckpoint() {
     // set an empty list to messages when deserialize
     private void readObject(java.io.ObjectInputStream stream)
         throws IOException, ClassNotFoundException {
+      stream.defaultReadObject();
       messages = new ArrayList<>();
```

Comment:
Since we override `messages` when we read the object, we should make `messages` as a `transient` field.

Reply:
It looks like it already was, doh.